packages/arquero-react/src/ArqueroDetailsList/__tests__/RowGroupingTestStory/RowGroupingTestStory.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ColumnMetadata, TableMetadata } from '@essex/arquero'
+import { introspect } from '@essex/arquero'
+import { ArqueroDetailsList, ArqueroTableHeader } from '@essex/arquero-react'
+import type { IDetailsGroupDividerProps } from '@fluentui/react'
+import { Label, PrimaryButton } from '@fluentui/react'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+import { memo, useCallback, useEffect, useState } from 'react'
+
+import { createLazyLoadingGroupHeader } from '../component-factories.js'
+import { ButtonContainer, Table } from './RowGroupingTestStory.styles.js'
+
+export interface RowGroupingTestStoryProps {
+	mockTable: ColumnTable | undefined
+}
+
+export const RowGroupingTestStory: React.FC<RowGroupingTestStoryProps> = memo(
+	function RowGroupingTestStory({ mockTable }) {
+		const [groupedTable, setGroupedTable] = useState<ColumnTable | undefined>()
+		const [groupedMetadata, setGroupedMetadata] = useState<
+			TableMetadata | undefined
+		>()
+
+		const [groupBy, setGroupBy] = useState<string>('')
+
+		useEffect(() => {
+			if (mockTable !== undefined) {
+				const mockTableCopy = mockTable
+				groupBy === ''
```

Comment:
With the groupby default undefined, you can reverse and simplify this logic to just be:

```
const mockTableCopy = groupby ? mockTableCopy.groupby([groupby]) : mockTable
setGroupedTable(mockTableCopy)
```

Reply:
If you switch it to an array, the logic would instead be something like `groupby.length > 0 ? mockTableCopy.groupby(groupby)...`